pkg/server/server_test.go

```diff
@@ -1199,16 +1140,10 @@ func TestServerDbOperations(t *testing.T) {
 }
 
 func TestServerUpdateConfigItem(t *testing.T) {
-	dataDir := "test-server-update-config-item-config"
+	dataDir := filepath.Join(t.TempDir(), "test-server-update-config-item-config")
 	configFile := fmt.Sprintf("%s.toml", dataDir)
-	s, closer := testServer(DefaultOptions().
-		WithAuth(false).
-		WithMaintenance(false).
-		WithDir(dataDir))
-	defer func() {
-		closer()
-		os.Remove(configFile)
-	}()
+	s, closer := testServer(DefaultOptions().WithAuth(false).WithMaintenance(false).WithDir(dataDir))
+	defer closer()
```

Comment:
No `t.Cleanup` here?

Reply:
The `testServer` method will return the `closer` functor just in case you'd like to close the server before the test finishes (IIRC we have some cases for that). Then since we've got the `closer` anyway, it looks more idiomatic to me to just use `defer` here.